osquery/tables/system/windows/certificates.cpp

```diff
@@ -307,7 +337,236 @@ void parseSystemStoreString(LPCWSTR sysStoreW,
+  std::string subjectKeyId;
+  toHexStr(certBuff.begin(), certBuff.end(), subjectKeyId);
+  r["subject_key_id"] = subjectKeyId;
+
+  r["path"] =
+      storeLocation + "\\" + constructDisplayStoreName(storeId, storeName);
+  r["store_location"] = storeLocation;
+  r["store"] = storeName;
+
+  std::string serial;
+  toHexStr(certContext->pCertInfo->SerialNumber.pbData,
+           certContext->pCertInfo->SerialNumber.pbData +
+               certContext->pCertInfo->SerialNumber.cbData,
+           serial);
+  r["serial"] = serial;
+
+  std::string authKeyId;
+  if (certContext->pCertInfo->cExtension != 0) {
+    auto extension = CertFindExtension(szOID_AUTHORITY_KEY_IDENTIFIER2,
+                                       certContext->pCertInfo->cExtension,
+                                       certContext->pCertInfo->rgExtension);
+    if (extension != nullptr) {
+      unsigned long decodedBuffSize = 0;
+      CryptDecodeObjectEx(CERT_ENCODING,
+                          X509_AUTHORITY_KEY_ID2,
+                          extension->Value.pbData,
+                          extension->Value.cbData,
+                          CRYPT_DECODE_NOCOPY_FLAG,
+                          nullptr,
+                          nullptr,
+                          &decodedBuffSize);
+
+      certBuff.resize(decodedBuffSize, 0);
+      std::fill(certBuff.begin(), certBuff.end(), 0);
+      auto decodeRet = CryptDecodeObjectEx(CERT_ENCODING,
+                                           X509_AUTHORITY_KEY_ID2,
+                                           extension->Value.pbData,
+                                           extension->Value.cbData,
+                                           CRYPT_DECODE_NOCOPY_FLAG,
+                                           nullptr,
+                                           certBuff.data(),
+                                           &decodedBuffSize);
+      if (decodeRet != FALSE) {
+        auto authKeyIdBlob =
+            reinterpret_cast<CERT_AUTHORITY_KEY_ID2_INFO*>(certBuff.data());
+
+        toHexStr(authKeyIdBlob->KeyId.pbData,
+                 authKeyIdBlob->KeyId.pbData + authKeyIdBlob->KeyId.cbData,
+                 authKeyId);
+      } else {
+        VLOG(1) << "Failed to decode authority_key_id with (" << GetLastError()
+                << ")";
+      }
+    }
+  }
+  r["authority_key_id"] = authKeyId;
+
+  results.push_back(r);
+}
+
+void findUserPersonalCertsOnDisk(const std::string& username,
+                                 const std::string& storeId,
+                                 const std::string& sid,
+                                 const std::string& storeName,
+                                 const std::string& storeLocation,
+                                 QueryData& results) {
+  std::stringstream certsPath;
+  certsPath
+      << getSystemRoot().root_name().string() << "\\Users\\" << username
+      << "\\AppData\\Roaming\\Microsoft\\SystemCertificates\\My\\Certificates";
+
+  try {
```

Comment:
nit: I'll leave this one up to you, but this `try` is encompassing quite a few API calls. What are you `try`ing to catch here? My concern is that if an exception is thrown in `getEncodedCert`, `CertCreateCertificateContext`, `addCertRow`, or in your simple enumeration of the file system, we'll always see the same error message displayed. Could this `try`/`catch` be brought in a bit closer to the parts where you think an issue might occur?

Reply:
The `try` is meant for `fs::directory_iterator` which will throw `filesystem_error`. This is mainly to catch when a nonexistent `certsPath` is passed in which can occasionally happen. Since only `filesystem_error` is caught, exceptions raised from the functions you mentioned shouldn't trigger this logging message (and will either crash the program, or bubble up to a higher catch). Because of how we use `directory_iterator` in the for loop, I'm not sure the scope can be tighter.

https://www.boost.org/doc/libs/1_70_0/libs/filesystem/doc/reference.html#Error-reporting